src/Xamarin.Android.Build.Tasks/Resources/StubApplication.java

```diff
@@ -305,25 +305,38 @@ private void monkeyPatchExistingResources() {
       mGetInstance.setAccessible(true);
       Object resourcesManager = mGetInstance.invoke(null);
 
-      Field mAssets = Resources.class.getDeclaredField("mAssets");
-      mAssets.setAccessible(true);
-
-      // Iterate over all known Resources objects
-      Field fMActiveResources = clazz.getDeclaredField("mActiveResources");
-      fMActiveResources.setAccessible(true);
-      @SuppressWarnings("unchecked")
-      Map<?, WeakReference<Resources>> arrayMap =
-          (Map<?, WeakReference<Resources>>) fMActiveResources.get(resourcesManager);
-      for (WeakReference<Resources> wr : arrayMap.values()) {
-        Resources resources = wr.get();
-        // Set the AssetManager of the Resources instance to our brand new one
-        mAssets.set(resources, newAssetManager);
-        resources.updateConfiguration(resources.getConfiguration(), resources.getDisplayMetrics());
+      Field mAssets = null;
+      try {
+          // Android N and later has mResourcesImpl/*of ResourcesImpl*/.mAssets instead.
+          Class<?> resImplClass = Class.forName("android.content.res.ResourcesImpl");
+          mAssets = resImplClass.getDeclaredField("mAssets");
+      } catch (NoSuchFieldException ex) {
+          try {
+	          mAssets = Resources.class.getDeclaredField("mAssets");
```

Comment:
Indentation is "weird" here -- indentation doesn't match the following line.. Is there a mixed tabs+spaces issue?